lib/block-supports/typography.php

```diff
@@ -276,31 +276,30 @@ function gutenberg_get_typography_value_and_unit( $raw_value, $options = array()
 		return null;
 	}
 
-	// Converts numeric values to pixel values by default.
-	if ( is_numeric( $raw_value ) ) {
-		$raw_value = $raw_value . 'px';
-	}
-
 	$defaults = array(
 		'coerce_to'        => '',
 		'root_size_value'  => 16,
 		'acceptable_units' => array( 'rem', 'px', 'em' ),
 	);
-
-	$options = wp_parse_args( $options, $defaults );
+	$options  = wp_parse_args( $options, $defaults );
 
 	$acceptable_units_group = implode( '|', $options['acceptable_units'] );
-	$pattern                = '/^(\d*\.?\d+)(' . $acceptable_units_group . '){1,1}$/';
+	$pattern                = '/^(-?\d*[\.,]?\d+)(' . $acceptable_units_group . '){0,1}$/';
```

Comment:
Now accepts negative numbers. This was an unrelated limitation.

`-10rem`

Also now accepts any pre-stringified floats from other locales:

`12,232px`

And unit less values:

`10.333` to which a `px` unit will be added by default. This matches existing behaviour only that we move it down to after the regex matching so we can do the string replace on any found values. Note that incoming `$raw_value` can be anything, e.g., `clamp(...)` or whatever, so I'm letting the regex do the check before we get clever.